File: 2020/day3/part1/main.go
package part1

import (
	"advent-of-code/advent_of_code/helpers"
	"fmt"
)

func Loop(lengthOfRow int, index int, row string) bool {
	pos := row[index%lengthOfRow]

	if string(pos) == "#" {
		return true
	}
	return false
}

func Main(file string) int {

	inputFile, err := helpers.ReadFileLineByLine(file)
	if err != nil {
		fmt.Println("%n", err)
	}
	var count int
	var index int

	for _, row := range inputFile {
		res := Loop(len(row), index, row)
		if res {
			count++
		}
		index += 3
	}

	return count

}

File: 2020/day3/part1/main.py
from advent_of_code.helpers.helpers import read_file_line_by_line

from typing import List


def loop(length_of_row: int, index: int, row: List[str]) -> bool:
    pos: str = row[index % length_of_row]  # maths x mod n for circular

    if pos == "#":
        return True
    return False


def main(file: str = "") -> int:

    input_file: List[str] = read_file_line_by_line(filepath=file)
    count: int = 0
    index: int = 0

    for row in input_file:
        res: int = loop(len(row), index=index, row=row)
        count += 1 if res else 0
        index += 3

    return count
